core-codemods/src/main/java/io/codemodder/codemods/SensitiveDataLoggingCodemod.java

```diff
@@ -1,53 +1,205 @@
+
+      if (statement.isPresent()) {
+        SensitivityAndFixAnalysis analysis;
+        try {
+          analysis = performSensitivityAnalysis(rawLines, startLine);
+        } catch (IOException e) {
+          throw new UncheckedIOException("Couldn't perform sensitivity analysis", e);
+        }
+        if (analysis.isSensitiveAndDirectlyLogged()) {
+          String newStatement = analysis.newStatement();
+          if (newStatement != null && !newStatement.isBlank()) {
+            Statement newStmt = StaticJavaParser.parseStatement(newStatement);
+            statement.get().replace(newStmt);
+
+            String analysisText = analysis.isSensitiveAnalysisText();
+            CodemodChange change = CodemodChange.from(startLine, analysisText);
+            changes.add(change);
+          }
+        }
+      }
+    }
+    return CodemodFileScanningResult.from(changes, List.of());
   }
 
-  @Override
-  protected String getFixPrompt() {
-    return getClassResourceAsString(getClass(), "fix_prompt.txt");
+  private SensitivityAndFixAnalysis performSensitivityAnalysis(
+      final List<String> lines, final Integer startLine) throws IOException {
+    String codeSnippet = numberedContextWithExtraLines(lines, startLine, 10, 10);
+    String prompt =
+        """
+              A tool has cited line %d of the code for possibly logging sensitive data:
+
+              %s
+
+              Respond ONLY in the form of JSON with the following keys, in this order:
+
+              sensitive_analysis_text: a careful, thorough analysis of whether the data is sensitive (specifically a password, session ID, security token, SSN, etc -- not a username)
+              is_data_directly_logged: a careful, thorough analysis of whether the data is definitely and directly logged (e.g., not just passed to another method inside to the scope, unless that's a method that obviously returns the given string)
+              is_it_sensitive_and_directly_logged: a boolean dictating whether it is sensitive and definitely and directly logged
+              new_line_to_replace: if sensitive and directly logged, the statement on line %d that should replace it -- remember to correctly JSON escape this value
+              """
+            .formatted(startLine, codeSnippet, startLine);
+
+    ChatCompletionRequest request =
+        ChatCompletionRequest.builder()
+            .temperature(0D)
+            .model("gpt-4o-2024-05-13")
+            //            .model("gpt-4-0613")
+            .n(1)
+            .messages(List.of(new ChatMessage(ChatMessageRole.USER.value(), prompt)))
+            .build();
+    ChatCompletionResult completion = service.createChatCompletion(request);
+    ChatCompletionChoice response = completion.getChoices().get(0);
+    String responseText = response.getMessage().getContent();
+    if (responseText.startsWith("```json") && responseText.endsWith("```")) {
+      responseText = responseText.substring(7, responseText.length() - 3);
```

Comment:
For clarity

```suggestion
      responseText = responseText.substring("```json".length(), responseText.length() - "```".length());
```